cardinal/server/server.go

```diff
@@ -76,19 +71,41 @@ func (s *Server) Serve() error {
 	if err != nil {
 		return eris.Wrap(err, "error getting hostname")
 	}
+
+	// Start server
 	log.Info().Msgf("serving at %s:%s", hostname, s.config.port)
-	s.isRunning.Store(true)
 	err = s.app.Listen(":" + s.config.port)
 	if err != nil {
 		return eris.Wrap(err, "error starting Fiber app")
 	}
-	s.isRunning.Store(false)
+
+	return nil
+}
+
+func (s *Server) BroadcastEvent(event any) error {
```

Comment:
this name feels a bit misleading; reading this code id assume we are broadcasting a singular event, but its only usage is broadcasting tick results which contains multiple events + some other data
```
type TickResults struct {
    Tick     uint64
    Receipts []receipt.Receipt
    Events   [][]byte
}
```

unless the intent is that `TickResults` is itself to be considered an `event`. unsure! but seems a bit quirky

Reply:
im equally unsure, im just following what was there. i say we just keep it this way then revisit

Reply:
Anything json serializeable is an event. The eventHub was originally agnostic to what a tick is or what the game engine is, it's just something that broadcasts events. 

I agree with @smsunarto keeping the event hub as it is allows for future extension. EventHub right now is only used to send TickResults over one websocket connection, but it supports much more. Well it's not an eventHub anymore but the intent is to allow the broadcasting of anything.